File: android/src/main/java/dev/barooni/capacitor/calendar/models/inputs/DeleteEventWithPromptInput.kt
package dev.barooni.capacitor.calendar.models.inputs

import com.getcapacitor.PluginCall
import dev.barooni.capacitor.calendar.PluginError
import dev.barooni.capacitor.calendar.models.enums.EventSpan

data class DeleteEventWithPromptInput(
    private val call: PluginCall,
) {
    val id: Long = call.getString("id")?.toLongOrNull() ?: throw PluginError.MissingId
    val title: String = call.getString("title") ?: throw PluginError.TitleMissing
    val message: String = call.getString("message") ?: throw PluginError.MessageMissing
    val confirmButtonText: String = call.getString("confirmButtonText") ?: "Delete"
    val cancelButtonText: String = call.getString("cancelButtonText") ?: "Cancel"
    val span: EventSpan = call.getInt("span")?.let { EventSpan.fromInt(it) } ?: EventSpan.THIS_EVENT
}

File: ios/Sources/CapacitorCalendarPlugin/Models/Inputs/DeleteEventWithPromptInput.swift
import Capacitor
import EventKit

struct DeleteEventWithPromptInput {
    private let id: String
    private let span: EKSpan
    private let title: String
    private let message: String
    private let confirmButtonText: String
    private let cancelButtonText: String

    init(call: CAPPluginCall) throws {
        guard let id = call.getString("id") else {
            throw PluginError.idMissing
        }
        self.id = id
        if let spanInt = call.getInt("span"), let span = EKSpan(rawValue: spanInt) {
            self.span = span
        } else {
            self.span = .thisEvent
        }
        guard let title = call.getString("title") else {
            throw PluginError.titleMissing
        }
        self.title = title
        guard let message = call.getString("message") else {
            throw PluginError.messageMissing
        }
        self.message = message
        self.confirmButtonText = call.getString("confirmButtonText", "Delete")
        self.cancelButtonText = call.getString("cancelButtonText", "Cancel")
    }

    func getId() -> String {
        return id
    }
    
    func getSpan() -> EKSpan {
        return span
    }
    
    func getTitle() -> String {
        return title
    }
    
    func getMessage() -> String {
        return message
    }
    
    func getConfirmButtonText() -> String {
        return confirmButtonText
    }
    
    func getCancelButtonText() -> String {
        return cancelButtonText
    }
}
